src/contracts.ts

```diff
@@ -18,33 +18,36 @@ export class Contracts {
 
     constructor(
         web3Instance: Web3,
-        chainId: number,
+        chainId?: number,
         contractAddresses?: RelayingServicesAddresses
     ) {
         this.web3Instance = web3Instance;
-        this.addresses =
-            contractAddresses ??
-            getContractAddresses(chainId ?? DEFAULT_NETWORK_ID);
-        this.initialize()
-            .then(() => {
-                console.debug('Contracts initialized correctly');
-            })
-            .catch((error) => {
-                console.error('Contracts fail to initialize', error);
-            });
-    }
-
-    async initialize(): Promise<void> {
-        this.smartWalletRelayVerifier = getContract(
-            this.web3Instance,
-            RelayVerifier.abi,
-            this.addresses.smartWalletRelayVerifier
+        contractAddresses = contractAddresses ?? <RelayingServicesAddresses>{};
+        const contracts: RelayingServicesAddresses = getContractAddresses(
+            chainId ?? DEFAULT_NETWORK_ID
         );
-        this.smartWalletDeployVerifier = getContract(
-            this.web3Instance,
-            DeployVerifier.abi,
-            this.addresses.smartWalletDeployVerifier
+        this.addresses = <RelayingServicesAddresses>(
+            mergeConfiguration(contractAddresses, contracts)
         );
+        this.initialize();
+    }
+
+    protected initialize(): void {
+        try {
+            this.smartWalletRelayVerifier = getContract(
+                this.web3Instance,
+                RelayVerifier.abi,
+                this.addresses.smartWalletRelayVerifier
+            );
+            this.smartWalletDeployVerifier = getContract(
+                this.web3Instance,
+                DeployVerifier.abi,
+                this.addresses.smartWalletDeployVerifier
+            );
+            console.debug('Contracts initialized correctly');
+        } catch (error) {
+            console.error('Contracts fail to initialize', error);
```

Comment:
It's not clear to me why we catch this error? It looks like we don't do anything more than `console.error`, but in that case, I would let the user decide what to do.

Reply:
Sorry, but in that case, we catch the error, but the client has no info about the error we caught. I would remove this catch clause to let the error arrive up to the user.